src/coldstorage-subaccount/ColdStorageHook.sol

```diff
@@ -199,18 +200,18 @@ contract ColdStorageHook is ERC7579HookDestruct {
         override
         returns (bytes memory hookData)
     {
-        bytes4 functionSig = bytes4(callData[0:4]);
-        console2.logBytes(callData);
+        bytes4 functionSig;
+
+        if (callData.length >= 4) {
+            functionSig = bytes4(callData[0:4]);
+        }
 
-        console2.log("request");
-        console2.log(target);
-        console2.logBytes4(functionSig);
-        console2.logBytes4(this.requestTimelockedExecution.selector);
         // check if call is a requestTimelockedExecution
         if (target == address(this) && functionSig == this.requestTimelockedExecution.selector) {
             return abi.encode(this.requestTimelockedExecution.selector);
         } else {
-            bytes32 executionHash = _execDigestMemory(target, value, callData);
+            bytes32 executionHash =
+                _execDigestMemory(target, value, callData[0:callData.length - 8]);
```

Comment:
@zeroknots for some reason the calldata is 8 bytes too long and if there is no calldata (ie eth transfer) then it is 12 zero bytes long

Reply:
this only works if you change `ExecutionLib.decodeSingle(msgData[36:])` to `ExecutionLib.decodeSingle(msgData[100:])` in `HookDestruct` in the ModuleKit

Reply:
so is msgData[100:] a fix then?

Reply:
no because we need the hack of `callData[0:callData.length - 8]` or `callData[0:callData.length - 12]`